models/packages/package_blob.go

```diff
@@ -98,3 +105,42 @@ func GetTotalUnreferencedBlobSize(ctx context.Context) (int64, error) {
 		Where("package_file.id IS NULL").
 		SumInt(&PackageBlob{}, "size")
 }
+
+// IsBlobAccessibleForUser tests if the user has access to the blob
+func IsBlobAccessibleForUser(ctx context.Context, blobID int64, user *user_model.User) (bool, error) {
```

Comment:
Could we also check `read` and `write` permission?

Reply:
This method does only check read access. Since blobs never change the is no reason for a write check. If a blob should be written, it does not exist, so nothing to check.